// @keystonehq/keystone-sdk/dist/tracker.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.track = exports.config = void 0;
const config_1 = require("./config");
exports.config = {
    enable: true,
    origin: '',
    xfp: '',
    mixpanel: {
        token: config_1.MIXPANEL_TOKEN,
        projectId: config_1.MIXPANEL_PROJECT_ID
    }
};
function track(event, payload) {
    var _a, _b;
    if (!exports.config.enable) {
        return;
    }
    const url = `https://api.mixpanel.com/import?strict=1&project_id=${exports.config.mixpanel.projectId}`;
    const xfp = (_a = payload.xfp) !== null && _a !== void 0 ? _a : exports.config.xfp;
    const time = (_b = payload.time) !== null && _b !== void 0 ? _b : Date.now();
    const insertId = typeof payload.requestId === 'string' ? payload.requestId : xfp;
    const body = JSON.stringify([
        {
            properties: {
                distinct_id: exports.config.origin,
                time,
                $insert_id: insertId,
                xfp
            },
            event
        }
    ]);
    const options = {
        method: 'POST',
        headers: {
            Accept: 'application/json',
            'Content-Type': 'application/json',
            Authorization: `Basic ${exports.config.mixpanel.token}`
        },
        body
    };
    fetch(url, options).catch(console.error);
}
exports.track = track;
